### ISSUE TITLE
Implement new importer state "partial_success"

### ISSUE THREAD

**Issue by Kartones** [CartoDB/cartodb#4502]
Currently, when you import something, the a single import might spawn N `Result` objects, but they only allow states `success`/`failure`, so when you have something that worked but had some errors (e.g. a zip file contained 3 csvs and 2 imported ok) there's no feedback that some data was lost. Or other times we go too radical and directly mark the import as failed.

Adding a third `partial_success` state and `user_feedback` field would allow to notify the user about those incomplete or partially errored imports.

As a rough estimation, this would need:
- [ ] New `Result` fields and state support at code (hardest part is going to be propagating the feedback stuff same as we do with the `Log` now)
- [ ] Import flow storage of feedback whenever is needed (e.g. some rows of a SHP get skipped due to encoding issues but others do import)
- [ ] Import Email changes?
- [ ] Sync tables also should support this state (maybe just treat it as success, but code will need changes for sure)
- [ ] Tests
- Frontend changes to show new info, etc.

@saleiva design this please and reassign to me so I can properly think about any additional details

**Comment by saleiva** [comment]
Hey @josecruz talk to me before starting with this

**Comment by josecruz** [comment]
Thoughts? Missing something? Not sure if we should show a modal:

![create - file importing partial error](https://cloud.githubusercontent.com/assets/596509/9086622/5f8222ac-3b7d-11e5-901a-396984b2d1ce.png)

![create - file importing partial error copy](https://cloud.githubusercontent.com/assets/596509/9086625/61b937c2-3b7d-11e5-9c96-238eb9bd782e.png)

**Comment by Kartones** [comment]
Uhm... You're the UX guys, but isn't too dangerous to leave errors in such a small area? if was me I'd go to showing a full modal (like when an import fully fails) and there show the errors.

Take into account we can import huge tables and thus have hundreds of failing rows due to encoding errors, or invalid geometries... things like that can span lots of verbose messages.

**Comment by josecruz** [comment]
Yep, talked about that with Sergio. Last version
![create - file importing partial error](https://cloud.githubusercontent.com/assets/596509/9093179/7196ca2c-3ba2-11e5-958e-71bd29607c05.png)

![create - file importing partial error - variation](https://cloud.githubusercontent.com/assets/596509/9093190/7a826c36-3ba2-11e5-8f94-7004ca90acef.png)

**Comment by saleiva** [comment]
I'm not sure I get the 2nd step...

2015-08-05 19:48 GMT+02:00 josecruz <notifications@github.com>:

> Yep, talked about that with Sergio. Last version
> [image: create - file importing partial error]
> <https://cloud.githubusercontent.com/assets/596509/9093179/7196ca2c-3ba2-11e5-958e-71bd29607c05.png>
>
> [image: create - file importing partial error - variation]
> <https://cloud.githubusercontent.com/assets/596509/9093190/7a826c36-3ba2-11e5-8f94-7004ca90acef.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CartoDB/cartodb/issues/4502#issuecomment-128089114>.
>



-- 
Sergio Alvarez Leiva

**Comment by josecruz** [comment]
this one?
<img width="310" alt="screen shot 2015-08-06 at 11 13 42" src="https://cloud.githubusercontent.com/assets/596509/9109259/38a1ea5a-3c2c-11e5-9d20-02d397fbc497.png">
just a placeholder

**Comment by Kartones** [comment]
remember... there can be hundreds of errors in that step `(2)`

**Comment by saleiva** [comment]
but what kind of things we can say there?

**Comment by Kartones** [comment]
Things like "Row 123 had a wrong, non UTF-8 encoding", "Layer xxxx had invalid geometries and couldn't be imported", etc., etc.

**Comment by josecruz** [comment]
uhmm they're not exactly steps, so I think that this modal could work right?

**Comment by josecruz** [comment]
up!

**Comment by saleiva** [comment]
Ok, I'd remove the part of opening in QGIS since that doesn't fix the problem.
I'd add istead of that something like 'Find the details about your errors below".

Since we can have multiple errors on 'step 2', how are we planning to show it? bullets? list?

**Comment by josecruz** [comment]
Okay so another take

![create - file importing partial error - variation](https://cloud.githubusercontent.com/assets/596509/9305675/7f50a804-44f3-11e5-8c81-96bc54e31022.png)

**Comment by saleiva** [comment]
Liking it much more, but think on what would happen if reporting 10 different errors...

**Comment by josecruz** [comment]
Could we add scroll?
![create - file importing partial error - variation copy](https://cloud.githubusercontent.com/assets/596509/9305810/85553c0a-44f4-11e5-84dc-fe3154644dc1.png)

**Comment by saleiva** [comment]
Pfff... nop.

If you have that amount of errors, you better say 'and 30 more...' don't you think?

**Comment by josecruz** [comment]
And where would you show the rest of them?

**Comment by saleiva** [comment]
I woulnd't... I would try to show the most representative ones...

**Comment by josecruz** [comment]
Okay, so:
![create - file importing partial error - variation copy](https://cloud.githubusercontent.com/assets/596509/9306022/f44cdee6-44f5-11e5-9e0e-7a763b72c4a7.png)

**Comment by Kartones** [comment]
Careful with that @saleiva . Either you provide me with a smart algorithm to show those "most representative ones" or I'll either show all or if I can't I'll just show the X first ones.

I understand that you might not like a scroll but the importer will just place inside a list all warnings and non-critical errors inside a list to show here, so we need to reach a middle-ground. 

Importer mostly works like a set of black boxes that communicate with one another (and many times only with one or two near them, to to all); also, as we move more and more logic down to SQL and the PG extension, we'll have even less control over the context of what happened, so I'd try to not be too smart guessing which errors are critical. 

I'd personaly go for showing the first X ones, then provide some way to check all of them (make the "and 20 more" clickable). Don't know how, but hiding some errors from the user doesn't helps a lot if you want to fix all of them at once...

**Comment by saleiva** [comment]
Taking in consideration this is an enhancement, I'd approach this in the simplest way, so we show the X first items and then the and XXX more indicator... 

Let's iterate!

**Comment by josecruz** [comment]
Okay so after talk with @saleiva we're going to iterate after we launch this modal, assigning back to @Kartones

**Comment by Kartones** [comment]
Ok, @saleiva your call at which milestone (I just cleared current one) for me to do the backend changes

**Comment by Kartones** [comment]
(I guess this now falls into Data-services, but if needed and I'm free, could implement it)

**Comment by alonsogarciapablo** [comment]
There has been no activity on this issue for more several months. We are closing it. If you think this still needs to be addressed please open a new issue.
